### PR TITLE
Fix API overview sidebars

### PR DESCRIPTION
**Pull request by wbamberg**
This PR fixes all the "web-api-overview" pages under Web/API which erroneously use "APIRef" instead of "DefaultAPISidebar".

Except for https://developer.mozilla.org/en-US/docs/Web/API/Constraint_validation.

### CODE REVIEW

**Comment by wbamberg** [comment]
I started https://github.com/mdn/mdn-community/discussions/291 for Constraint Validation.

File: files/en-us/web/api/push_api/index.md [hunk 1]
```diff
@@ -15,7 +15,7 @@ browser-compat:
   - api.PushMessageData
 ---
 
-{{ApiRef("Push API")}}
+{{DefaultAPISidebar("Push API")}}
```
Comment (bsmth) [inline files/en-us/web/api/push_api/index.md:18]: This one adds a horizontal scroll to the sidebar:

https://pr22468.content.dev.mdn.mozit.cloud/en-US/docs/Web/API/Push_API

**Comment by bsmth** [comment]
Aside from the Push API page, the preview pages look better. I had to [look up the macro](https://github.com/mdn/yari/blob/main/kumascript/macros/DefaultAPISidebar.ejs) to find out what it does, but I could still do with hints when to use this over other. Do we need to add anything to [the sidebar docs](https://developer.mozilla.org/en-US/docs/MDN/Writing_guidelines/Page_structures/Macros/Commonly_used_macros#sidebar_generation) for this?

The Prettier formatting looks good.

**Comment by wbamberg** [comment]
> Aside from the Push API page, the preview pages look better. I had to [look up the macro](https://github.com/mdn/yari/blob/main/kumascript/macros/DefaultAPISidebar.ejs) to find out what it does, but I could still do with hints when to use this over other. 

API overview pages (that is, pages with `page-type: web-api-overview`) should use `{{DefaultAPISidebar}}`, and all other pages should use `{{APIRef}}`.

A long time ago I wrote up a kind of guide to them: https://discourse.mozilla.org/t/defaultapisidebar-apiref-and-groupdata/40210 but I don't know if there is anything comparable on MDN.

> Do we need to add anything to [the sidebar docs](https://developer.mozilla.org/en-US/docs/MDN/Writing_guidelines/Page_structures/Macros/Commonly_used_macros#sidebar_generation) for this?
> 

Those docs are hopelessly inadequate, so probably. Sidebars for Web/API are ridiculously complicated. Ruth and I proposed a much simpler solution in https://github.com/openwebdocs/project/issues/76 but it doesn't seem to be going anywhere at the moment.

> The Prettier formatting looks good.

File: files/en-us/web/api/push_api/index.md (hunk 1 above)
Comment (wbamberg) [reply files/en-us/web/api/push_api/index.md:18]: > This one adds a horizontal scroll to the sidebar:
> 
> https://pr22468.content.dev.mdn.mozit.cloud/en-US/docs/Web/API/Push_API

That's a bug in the sidebar. Note that the current one does too: https://developer.mozilla.org/en-US/docs/Web/API/Push_API.

The current one is also broken in various ways:
- it lists the main overview page twice, the second time with the slug, code formatted, as the title
- it doesn't list properties or events of this API
- it doesn't list interfaces under "Interfaces"

Using `{{APIRef}}` is just an error on the part of the author, it's not a stylistic choice. It's not optional to`{{DefaultAPISidebar}}` for this type of page.

**Comment by bsmth** [comment]
Thank you for the explanation!

**Review by bsmth — APPROVED** [review]
Looks good, thank you! I'm happy to merge when you are.

**Pull request merged**